### PR TITLE
🌱 Update hetznerbaremetalhost_types.go

### PR DESCRIPTION
**Pull request by Sayanta66**
**What this PR does / why we need it**:

Added spec descriptions and fixes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [x] include documentation
- [ ] add unit tests

### CODE REVIEW

File: api/v1beta1/hetznerbaremetalhost_types.go [hunk 1]
```diff
@@ -46,11 +46,12 @@ const (
...
 // Currently, you can specify one disk or a raid setup.
 type RootDeviceHints struct {
-	// Unique storage identifier. The hint must match the actual value
-	// exactly.
+	// WWN is a unique storage identifier used for non-raid setups. The hint must match 
+	// the actual value exactly.
 	// +optional
 	WWN string `json:"wwn,omitempty"`
-	// To specify multiple storage devices.
+	// To specify multiple storage devices. It provides the controller with information
```
Comment (janiskemper) [inline api/v1beta1/hetznerbaremetalhost_types.go:53]: you have to start with "Raid"

**Review by janiskemper — CHANGES_REQUESTED** [review]

File: api/v1beta1/hetznerbaremetalhost_types.go [hunk 2]
```diff
@@ -74,6 +75,7 @@ func (rdh *RootDeviceHints) ListOfWWN() []string {
 
 // Raid can be used instead of WWN to point to multiple storage devices.
 type Raid struct {
+	// WWN defines a list of Unique storage identifiers used for raid setups.
```
Comment (janiskemper) [inline api/v1beta1/hetznerbaremetalhost_types.go:78]: Unique -> unique

**Review by janiskemper — CHANGES_REQUESTED** [review]

File: api/v1beta1/hetznerbaremetalhost_types.go (hunk 2 above)
Comment (janiskemper) [reply api/v1beta1/hetznerbaremetalhost_types.go:78]: this is not resolved

Comment (janiskemper) [reply api/v1beta1/hetznerbaremetalhost_types.go:78]: not solved

**Review by kranurag7 — DISMISSED** [review]

File: api/v1beta1/hetznerbaremetalhost_types.go [hunk 3]
```diff
@@ -121,24 +123,24 @@ const (
...
 	ErrorMessageMissingRescueSSHSecret string = "could not find RescueSSHSecret"
-	// ErrorMessageMissingOSSSHSecret specifies the error message when no OSSSH secret was found.
+	// ErrorMessageMissingOSSSHSecret specifies the error message when no OSSSH secret is found.
 	ErrorMessageMissingOSSSHSecret string = "could not find OSSSHSecret"
 	// ErrorMessageMissingOrInvalidSecretData specifies the error message when no data in secret is missing or invalid.
 	ErrorMessageMissingOrInvalidSecretData string = "invalid or not specified information in secret"
 )
 
-// ProvisioningState defines the states the provisioner will report the host has having.
+// ProvisioningState defines the states the provisioner will report the host was having.
```
Comment (janiskemper) [inline api/v1beta1/hetznerbaremetalhost_types.go:136]:
```suggestion
// ProvisioningState defines the states of provisioning of the host.
```

**Pull request merged**